src/meta/processors/indexMan/CreateEdgeIndexProcessor.cpp

```diff
@@ -44,6 +44,37 @@ void CreateEdgeIndexProcessor::process(const cpp2::CreateEdgeIndexReq& req) {
         return;
     }
 
+    auto prefix = MetaServiceUtils::indexPrefix(space);
+    std::unique_ptr<kvstore::KVIterator> checkIter;
+    auto checkRet = kvstore_->prefix(kDefaultSpaceId, kDefaultPartId, prefix, &checkIter);
+    if (checkRet != kvstore::ResultCode::SUCCEEDED) {
+        resp_.set_code(to(checkRet));
+        onFinished();
+        return;
+    }
+
+    while (checkIter->valid()) {
+        auto val = checkIter->val();
+        auto item = MetaServiceUtils::parseIndex(val);
+        if (item.get_schema_id().getType() != nebula::cpp2::SchemaID::Type::edge_type) {
+            continue;
+        }
+
+        for (size_t i = 0; i < fieldNames.size(); i++) {
+            if (fieldNames[i] != item.get_fields()[i].get_name()) {
```

Comment:
Do you ensure the item have the same numbers fields as the filedNames?

Reply:
In fact, when the index's field is disaccord, they are not the same.

Reply:
So maybe access invalid index on "item.get_fields()[i]"

Reply:
Add a checker, if the size of current index is longer than the previous one , it will skip.